3_javascript/javascript1/week2/script.js

```diff
@@ -0,0 +1,87 @@
+
+// exercice full_name
+function getFullname(firstname, surname, useFormalName, isWomen ){
+
+    if( firstname === "" & surname === ""){
+        return "firstname and surname are missing"
+    }
+    if (useFormalName === true){
```

Comment:
Since `useFormalName` and `isWoman` are boolean you don't need the ===true comparison to check if they are true. You can simply call it like this 
```
if(useFormalName && isWoman)
{
...
}
```
Also the empty string `""`is a [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) value in javascript.So you don't need the comparison there either because they are converted to boolean and will be false. 
So we can have something like this:

```javascript
// Exercise: Full Name
function getFullname(firstname, surname, useFormalName, isWoman) {
    // Check if both firstname and surname are missing
    if (!firstname || !surname) {
        return "Firstname and surname are missing";
    }

    // Check if useFormalName is true
    if (useFormalName) {
        // Check if isWoman is true
        if (isWoman) {
            return `Lady ${firstname} ${surname}`;
        } else {
            return `Lord ${firstname} ${surname}`;
        }
    } else {
        // If useFormalName is false, return the informal full name
        return `${firstname} ${surname}`;
    }
}
```